owncloudApp/src/main/java/com/owncloud/android/MainApp.kt

```diff
@@ -0,0 +1,277 @@
+                PassCodeManager.getPassCodeManager().onActivityStarted(activity)
+                PatternManager.getPatternManager().onActivityStarted(activity)
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    FingerprintManager.getFingerprintManager(activity).onActivityStarted(activity)
+                }
+            }
+
+            override fun onActivityResumed(activity: Activity) {
+                Log_OC.d(activity.javaClass.simpleName, "onResume() starting")
+            }
+
+            override fun onActivityPaused(activity: Activity) {
+                Log_OC.d(activity.javaClass.simpleName, "onPause() ending")
+            }
+
+            override fun onActivityStopped(activity: Activity) {
+                Log_OC.d(activity.javaClass.simpleName, "onStop() ending")
+                PassCodeManager.getPassCodeManager().onActivityStopped(activity)
+                PatternManager.getPatternManager().onActivityStopped(activity)
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    FingerprintManager.getFingerprintManager(activity).onActivityStopped(activity)
+                }
+                if (activity is PassCodeActivity ||
+                    activity is PatternLockActivity ||
+                    activity is FingerprintActivity
+                ) {
+                    WhatsNewActivity.runIfNeeded(activity)
+                }
+            }
+
+            override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {
+                Log_OC.d(activity.javaClass.simpleName, "onSaveInstanceState(Bundle) starting")
+            }
+
+            override fun onActivityDestroyed(activity: Activity) {
+                Log_OC.d(activity.javaClass.simpleName, "onDestroy() ending")
+            }
+        })
+
+        val newArchModule = module {
+            viewModel { (filePath: String, account: Account, shareTypes: List<ShareType>) ->
+                OCShareViewModel(filePath, account, shareTypes)
+            }
+
+            viewModel { (account: Account) ->
+                OCCapabilityViewModel(account)
+            }
+        }
+
+        startKoin {
+            modules(newArchModule)
+        }
+    }
+
+    companion object {
+
+        private val TAG = MainApp::class.java.simpleName
+
+        private val AUTH_ON = "on"
+
+        private val POLICY_SINGLE_SESSION_PER_ACCOUNT = "single session per account"
+        private val POLICY_ALWAYS_NEW_CLIENT = "always new client"
+
+        var appContext: Context? = null
+            private set
+
+        var BETA_VERSION = "beta"
+
+        /**
+         * Next methods give access in code to some constants that need to be defined in string resources to be referred
+         * in AndroidManifest.xml file or other xml resource files; or that need to be easy to modify in build time.
+         */
+
+        val accountType: String
```

Comment:
instead of having all these attributes here, why don't you create a class called ConfigParams whose constructor only have one item (context) and include this class as a dependency? For backward compatibility, I'd add koin java for those non-kotlin files.

Reply:
That change would be fine but I would not like to address it here, otherwise we won't never finish the architectural changes. Transforming this class to Kotlin is enough by now.

Reply:
ok...